kpatch-build/create-diff-object.c

```diff
@@ -3253,6 +3253,42 @@ static void kpatch_build_strings_section_data(struct kpatch_elf *kelf)
 	}
 }
 
+/*
+ * Don't allow sibling calls from patched functions on ppc64le.  Before doing a
+ * sibling call, the patched function restores the stack to its caller's stack.
+ * The kernel-generated stub then writes the patch module's r2 (toc) value to
+ * the caller's stack, corrupting it, eventually causing a panic after it
+ * returns to the caller and the caller tries to use the livepatch module's toc
+ * value.
+ *
+ * In theory we could instead a) generate a custom stub, or b) modify the
+ * kernel livepatch_handler code to save/restore the stack r2 value, but this
+ * is easier for now.
+ */
+static void kpatch_no_sibling_calls_ppc64le(struct kpatch_elf *kelf)
+{
+#ifdef __powerpc64__
+	struct symbol *sym;
+	void *insn;
+
+	list_for_each_entry(sym, &kelf->symbols, list) {
+		if (sym->type != STT_FUNC || sym->status != CHANGED)
+			continue;
+
+		for (insn = sym->sec->data->d_buf;
+		     insn < sym->sec->data->d_buf + sym->sec->data->d_size;
+		     insn += 4) {
+			if (*(unsigned int *)insn != 0x48000000)
```

Comment:
Endianness shouldn't matter, so treating it as an unsigned int is okay, I think, rather than a series of bytes.  However 0x48000000 looks like a magic number.. can we make this a preprocessor constant or at least add a comment to what this pattern is?